tests/psbt.rs

```diff
@@ -0,0 +1,704 @@
+use bitcoin::consensus::encode::{serialize_hex, deserialize};
+use bitcoin::blockdata::{opcodes, script};
+use opcodes::all::OP_PUSHBYTES_0 as OP_0;
+use bitcoin::hashes::Hash;
+use bitcoin::hashes::hex::{FromHex};
+use bitcoin::secp256k1::{self, Message, SECP256K1};
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey, Fingerprint, IntoDerivationPath, KeySource};
+use bitcoin::util::psbt::{PartiallySignedTransaction, PsbtSighashType};
+use bitcoin::util::sighash::SighashCache;
+
+const NETWORK: Network = Network::Testnet;
+
+type Psbt = PartiallySignedTransaction; // TODO: Consider adding this alias to `rust-bitcoin`.
+
+macro_rules! hex_script {
+    ($s:expr) => (<Script as FromStr>::from_str($s).unwrap());
+}
+
+macro_rules! hex_psbt {
+    ($s:expr) => (deserialize::<Psbt>(&<Vec<u8> as FromHex>::from_hex($s).unwrap()));
+}
+
+#[test]
+fn bip174_psbt_workflow() {
+    let secp = SECP256K1;
+
+    //
+    // Step 0: Create the extended private key from the test vector data.
+    //
+
+    let ext_priv = build_extended_private_key();
+    let ext_pub = ExtendedPubKey::from_priv(&secp, &ext_priv);
+    let fingerprint = ext_pub.fingerprint();
+
+    //
+    // Step 1: The creator.
+    //
+
+    let tx = create_transaction();
+    let psbt = create_psbt(tx);
+
+    //
+    // Step 2: The first updater.
+    //
+
+    // FIXME: This update implies the updater knows the extended public key,
+    // is that correct? Otherwise where does the fingerprint come from?
+    let psbt = update_psbt(psbt, fingerprint);
+
+    //
+    // Step 3: The second updater.
+    //
+
+    let psbt = update_psbt_with_sighash_all(psbt);
+
+    //
+    // Step 4: The first signer.
+    //
+
+    // Strings from BIP174 test vector.
+    let test_vector = vec![
+        ("cP53pDbR5WtAD8dYAW9hhTjuvvTVaEiQBdrz9XPrgLBeRFiyCbQr", "m/0h/0h/0h"),
+        ("cR6SXDoyfQrcp4piaiHE97Rsgta9mNhGTen9XeonVgwsh4iSgw6d", "m/0h/0h/2h"),
+    ];
+
+    // We pass the keys to the signer after doing verification to make explicit
+    // that signer is only using these two keys.
+    let keys = parse_and_verify_keys(&ext_priv, &test_vector);
+    let psbt_1 = signer_one_sign(psbt.clone(), &keys);
+
+    //
+    // Step 5: The second signer.
+    //
+
+    // Strings from BIP174 test vector.
+    let test_vector = vec![
+        ("cT7J9YpCwY3AVRFSjN6ukeEeWY6mhpbJPxRaDaP5QTdygQRxP9Au", "m/0h/0h/1h"),
+        ("cNBc3SWUip9PPm1GjRoLEJT6T41iNzCYtD7qro84FMnM5zEqeJsE", "m/0h/0h/3h"),
+    ];
+
+    let keys = parse_and_verify_keys(&ext_priv, &test_vector);
+    let psbt_2 = signer_two_sign(psbt, &keys);
+
+    //
+    // Step 6: Combiner the two signed PSBTs.
+    //
+
+    let combined = combine(psbt_1, psbt_2);
+
+    //
+    // Step 7: Finalize the PSBT.
+    //
+
+    let finalized = finalize(combined);
+
+    //
+    // Step 8: Extract the transaction.
+    //
+
+    let _tx = extract_transaction(finalized);
+
+    //
+    // Step 9: Test lexicographical PSBT combiner
```

Comment:
This is confusing as it implies that combining is the last step. If we are trying to test something else, we should create a separate test.

Reply:
Fair point, shall we remove this then @DanGould?

Reply:
Shall we? The test is the bip174 defined workflow, of which the lexicographical psbt combiner is a part

Reply:
Perhaps we could add a comment explicitly stating that sorting is not part of a typical workflow but is done here for testing purposes?

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/blob/e557e896dac9579ffc38da3656dbc85ac4820683/tests/psbt.rs#L132-L139

Reply:
nit: Can you please remove the `Step 9` part from this?

Reply:
> Shall we? The test is the bip174 defined workflow, of which the lexicographical psbt combiner is a part

Do you disagree with  this statement @sanket1729? Or is it that the newly added comment does not fully explain why the `combine_lexicagraphically` call is there?